HW_9/task_2.ts

```diff
@@ -0,0 +1,135 @@
+/*
+1. Создайте интерфейс IEmployee с полями name, surname, salary, address (типы для этих полей такие же как в ItEmployee из таск 1)
+  Создайте функцию getEmployeeInfo(employee), выводящую в консоль всю информацию про employee (формат текста придумать самим)
+  Создайте type guard isItEmployee. Используйте его в функции getEmployeeInfo.
+  Если на входе itEmployee - выводите в консоль все поля айтишника (а не только те, что в employee)
+  Функция должна принимать union type между Employee и ItEmployee, и через тайпгвард определять что за объект и как работать с ним
+*/
+
+// task_2.ts
+import { IItEmployee, IAddress, object } from './task_1';
+
+interface IEmployee {
+    name: string;
+    surname: string;
+    salary: number;
+    address: IAddress;
+}
+
+function isItEmployee(employee: IEmployee | IItEmployee): employee is IItEmployee {
+    return 'grade' in employee;
+}
+
+function getEmployeeInfo(employee: IEmployee | IItEmployee): string {
+    if (isItEmployee(employee)) {
+        return `IT Employee:
+            Name: ${employee.name}
+            Surname: ${employee.surname}
+            Salary: ${employee.salary}
+            Grade: ${employee.grade}
+            Occupation: ${employee.occupation}
+            Address: ${employee.address ? `${employee.address.country}, ${employee.address.street}, ${employee.address.house}, ${employee.address.flat}` : 'N/A'}
+            Projects: ${employee.projectNames.join(', ')}
+        `;
+    } else {
+        return `Employee:
+            Name: ${employee.name}
+            Surname: ${employee.surname}
+            Salary: ${employee.salary}
+            Address: ${employee.address.country}, ${employee.address.street}, ${employee.address.house}, ${employee.address.flat}
+        `;
+    }
+};
+
+
+const employee: IEmployee = {
+    name: 'Loh',
+    surname: 'Doe',
+    salary: 0,
+    address: {
+        country: 'Russia',
+        street: 'street',
+        house: 1,
+        flat: 2
+    }
+};
+
+
+
+console.log(getEmployeeInfo(employee));
+console.log(getEmployeeInfo(object));
+
+// Создайте функцию, которая подсчитает, сколько в объекте значений каждого типа.
+//   Принимает на вход объект или массив таких же объектов, у которого ключ всегда string, а значение - string, number, boolean.
+//   Возвращает же - объект с ключами string, number, boolean и количеством таких значений в объекте или в сумме у всех объектов в массиве.
+
+type TValueTypes = 'string' | 'number' | 'boolean';
+function countValueTypes(input: Record<string, string | number | boolean> | Array<Record<string, string | number | boolean>>): Record<TValueTypes, number> {

```

Comment:
You can create type alias for Recors<string, string | number | boolean> like CountValueTypesObject and use like 
countValueTypes(input: CountValueTypesObject | CountValueTypesObject[])